actiontext/lib/action_text/attribute.rb

```diff
@@ -56,12 +59,26 @@ def #{name}
           def #{name}?
             rich_text_#{name}.present?
           end
-
-          def #{name}=(body)
-            self.#{name}.body = body
-          end
         CODE
 
+        if store_if_blank
+          class_eval <<-CODE, __FILE__, __LINE__ + 1
+            def #{name}=(body)
+              self.#{name}.body = body
+            end
+          CODE
+        else
+          class_eval <<-CODE, __FILE__, __LINE__ + 1
+            def #{name}=(body)
+              if body.present?
+                self.#{name}.body = body
+              else
+                self.#{name}.mark_for_destruction if #{name}?
```

Comment:
Is the attachment record already created at this point? Could we prevent it from being created in the first place?

Reply:
This is for the case where you have a value persisted, and then it’s updated to an empty string.